packages/edit-widgets/src/components/sidebar/index.js

```diff
@@ -37,32 +37,104 @@ const WIDGET_AREAS_IDENTIFIER = 'edit-widgets/block-areas';
  */
 import WidgetAreas from './widget-areas';
 import { store as editWidgetsStore } from '../../store';
+import { unlock } from '../../lock-unlock';
+
+const { Tabs } = unlock( componentsPrivateApis );
+
+function SidebarHeader( { selectedWidgetAreaBlock } ) {
+	return (
+		<Tabs.TabList>
+			<Tabs.Tab tabId={ WIDGET_AREAS_IDENTIFIER }>
+				{ selectedWidgetAreaBlock
+					? selectedWidgetAreaBlock.attributes.name
+					: __( 'Widget Areas' ) }
+			</Tabs.Tab>
+			<Tabs.Tab tabId={ BLOCK_INSPECTOR_IDENTIFIER }>
+				{ __( 'Block' ) }
+			</Tabs.Tab>
+		</Tabs.TabList>
+	);
+}
 
-function ComplementaryAreaTab( { identifier, label, isActive } ) {
+function SidebarContent( {
+	hasSelectedNonAreaBlock,
+	currentArea,
+	isGeneralSidebarOpen,
+	selectedWidgetAreaBlock,
+} ) {
 	const { enableComplementaryArea } = useDispatch( interfaceStore );
+
+	// currentArea, and isGeneralSidebarOpen are intentionally left out from the dependencies,
+	// because we want to run the effect when a block is selected/unselected and not when the sidebar state changes.
+	useEffect( () => {
+		if (
+			hasSelectedNonAreaBlock &&
+			currentArea === WIDGET_AREAS_IDENTIFIER &&
+			isGeneralSidebarOpen
+		) {
+			enableComplementaryArea(
+				'core/edit-widgets',
+				BLOCK_INSPECTOR_IDENTIFIER
+			);
+		}
+		if (
+			! hasSelectedNonAreaBlock &&
+			currentArea === BLOCK_INSPECTOR_IDENTIFIER &&
+			isGeneralSidebarOpen
+		) {
+			enableComplementaryArea(
+				'core/edit-widgets',
+				WIDGET_AREAS_IDENTIFIER
+			);
+		}
+	}, [ hasSelectedNonAreaBlock, enableComplementaryArea ] );
+
+	const tabsContextValue = useContext( Tabs.Context );
+
 	return (
-		<Button
-			onClick={ () =>
-				enableComplementaryArea( editWidgetsStore.name, identifier )
-			}
-			className={ classnames( 'edit-widgets-sidebar__panel-tab', {
-				'is-active': isActive,
-			} ) }
-			aria-label={
-				isActive
-					? // translators: %s: sidebar label e.g: "Widget Areas".
-					  sprintf( __( '%s (selected)' ), label )
-					: label
+		<ComplementaryArea
+			className="edit-widgets-sidebar"
+			header={
+				<Tabs.Context.Provider value={ tabsContextValue }>
+					<SidebarHeader
+						selectedWidgetAreaBlock={ selectedWidgetAreaBlock }
+					/>
+				</Tabs.Context.Provider>
 			}
-			data-label={ label }
+			headerClassName="edit-widgets-sidebar__panel-tabs"
+			/* translators: button label text should, if possible, be under 16 characters. */
+			title={ __( 'Settings' ) }
+			closeLabel={ __( 'Close Settings' ) }
+			scope="core/edit-widgets"
+			identifier={ currentArea }
+			icon={ isRTL() ? drawerLeft : drawerRight }
+			isActiveByDefault={ SIDEBAR_ACTIVE_BY_DEFAULT }
 		>
-			{ label }
-		</Button>
+			<Tabs.Context.Provider value={ tabsContextValue }>
+				<Tabs.TabPanel tabId={ WIDGET_AREAS_IDENTIFIER }>
```

Comment:
The last part of the keyboard testing instructions mention:

> specifically, you're making sure it doesn't focus the sidebar panel itself

Currently, because the we're not specifying `focusable={false}`, pressing `Tab` does focus the tabpanel element.

We should either update testing instructions, or add the `focusable={false}` prop. I guess for coherency with what done in other PRs, we should go for the latter?

Reply:
Oh. wow 🤦

I slapped `focusable={false}` on the parent `Tabs` component like a dummy 😆 Thanks for spotting that, I'll get it properly applied.

Reply:
The root `Tabs` component still had the `focusable` prop assigned, I pushed a small commit to remove it